### ISSUE TITLE
DB duplication when syncing with OneDrive

### ISSUE THREAD

**Issue by madriani** [PhilippC/keepass2android#428]
I am using the 1.05d version of Keepass2Android Offline on Android 8.1 (May 2018 patch, Android One version).
I keep my kdbx file on my personal OneDrive, where I occasionally edit it from my PC using KeePass.
I use the Android app mostly to read the DB (through the Android file picker and the official OneDrive app), as I am more comfortable to make edits from the PC.

I noticed that, from time to time, after using the Android app, I find another copy of my DB in its folder, with a " 1" attached to its name (i.e., I see two copies of the DB: "DB-PSW.kdbx" and "DB-PSW 1.kdbx").

This gets annoying since, on the PC, I make edits to the original file (without the " 1"), while the DB used by the phone is that with the " 1".

Unfortunately, I did not manage to find a pattern of use that leads to the duplication, but I suspect it is related to when I use the Android app after doing some edits from the PC.
The only workaround I found was to re-download the DB from the phone after deleting the additional copy.

I am available to make tests if needed to solve the issue.

**Comment by PhilippC** [comment]
KP2A cannot control this when you use the offline version (which means that some other app is doing the acual sync). I suggest you switch to regular Keepass2Android, especially the version 1.07b-pre2 (just published to beta, see https://play.google.com/apps/testing/keepass2android.keepass2android). This version has a completely new OneDrive implementation which should cover all feature requests.

**Comment by madriani** [comment]
Hello PhilippC, thanks for your answer.
Actually, I did not notice any duplication in the last 6 months (and probably even more), so I will probably stay with the offline version, at least for a while. I suppose the bug was in the OneDrive app, that got quite a few updates in the last year.
Anyway, good to know that a new OneDrive implementation has been added to the regular version.
Thank you again; have a nice day.
